### PR TITLE
fix several reactions problem related https://github.com/Mobsya/aseba/issues/503

### PR DESCRIPTION
**Pull request by mbonani**

### CODE REVIEW

File: src/extensions/scratch3_thymio/aesl.js [hunk 1]
```diff
@@ -16,15 +16,22 @@ const eventsDefinition = [
...
+    {name: 'M_motors', fixed_size: 2},
     {name: 'Q_reset', fixed_size: 0},
-    {name: 'tap', fixed_size: 0}
+    {name: 'tap', fixed_size: 0},
+	{name: 'B_center', fixed_size: 0},
+	{name: 'B_forward', fixed_size: 0},
+	{name: 'B_backward', fixed_size: 0},
+	{name: 'B_left', fixed_size: 0},
+	{name: 'B_right', fixed_size: 0}
 ];
```
Comment (cor3ntin) [inline src/extensions/scratch3_thymio/aesl.js:36]: Indentation

File: src/extensions/scratch3_thymio/aesl.js [hunk 2]
```diff
@@ -70,7 +102,10 @@ onevent Q_reset
 onevent V_leds_bottom
   if event.args[0]==0 then
     call leds.bottom.left(event.args[1],event.args[2],event.args[3])
+  elseif event.args[0]==1 then
+    call leds.bottom.right(event.args[1],event.args[2],event.args[3])
```
Comment (cor3ntin) [inline src/extensions/scratch3_thymio/aesl.js:106]: Indentation

File: src/extensions/scratch3_thymio/index.js [hunk 3]
```diff
@@ -118,6 +118,12 @@ class Thymio {
 
         this.tap = false;
         this.useHorizontalLeds = false;
+		
+		this.whenButtonCenter = false;
+		this.whenButtonForward = false;
+		this.whenButtonBackward = false;
+		this.whenButtonLeft = false;
+		this.whenButtonRight = false;
```
Comment (cor3ntin) [inline src/extensions/scratch3_thymio/index.js:127]: Indentation

File: src/extensions/scratch3_thymio/index.js [hunk 4]
```diff
@@ -241,22 +262,23 @@ class Thymio {
         } else if (motor === 'right') {
             this.sendAction('M_motor_right', args);
         } else {
-            this.sendActions({M_motor_left: args, M_motor_right: args});
+			args.unshift(value);
```
Comment (cor3ntin) [inline src/extensions/scratch3_thymio/index.js]: Indentation

File: src/extensions/scratch3_thymio/index.js [hunk 5]
```diff
@@ -390,12 +412,13 @@ class Thymio {
         speed = parseInt(Math.abs(speed), 10);
         speed = parseInt(clamp(speed, Thymio.VMIN * 10 / 32, Thymio.VMAX * 10 / 32), 10);
 
-        if (angle === 0) {
+        if (angle === 0) {//FIXME not working 
             const largs = Array();
```
Comment (cor3ntin) [inline src/extensions/scratch3_thymio/index.js:416]: What's the bug ?

Comment (mbonani) [reply src/extensions/scratch3_thymio/index.js:416]: it seems that the origin idea was "if angle was zero to give only the speed to wheels" but the strange calculation do not provide something that works

**Review by cor3ntin — APPROVED** [review]

**Pull request merged**